ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -173,67 +184,38 @@ private void initReplDump(ASTNode ast) throws HiveException {
-                Tree tablePatternNode = tablesNode.getChild(child);
-                if (tablePatternNode.getType() == TOK_NULL) {
-                  LOG.error(ErrorMsg.REPL_INVALID_DB_OR_TABLE_PATTERN.getMsg());
-                  throw new SemanticException(ErrorMsg.REPL_INVALID_DB_OR_TABLE_PATTERN.getMsg());
-                }
-                tablesList.add(unescapeSQLString(tablePatternNode.getText()));
-              }
-              if (!tablesList.isEmpty()) {
-                if (listIdx == 0) {
-                  LOG.info("ReplScope: Set Included Tables List: {}", tablesList);
-                  replScope.setIncludedTablePatterns(tablesList);
-                } else {
-                  LOG.info("ReplScope: Set Excluded Tables List: {}", tablesList);
-                  replScope.setExcludedTablePatterns(tablesList);
-                }
-              }
-            }
-          }
+          setReplDumpTablesList(currNode);
           break;
         }
-        default: {
+        case TOK_FROM: {
           // TOK_FROM subtree
           Tree fromNode = currNode;
           eventFrom = Long.parseLong(PlanUtils.stripQuotes(fromNode.getChild(0).getText()));
-          // skip the first, which is always required
-          int numChild = 1;
-          while (numChild < fromNode.getChildCount()) {
-            if (fromNode.getChild(numChild).getType() == TOK_TO) {
+
+          // Skip the first, which is always required
+          int fromChildIdx = 1;
+          while (fromChildIdx < fromNode.getChildCount()) {
+            if (fromNode.getChild(fromChildIdx).getType() == TOK_TO) {
               eventTo =
-                      Long.parseLong(PlanUtils.stripQuotes(fromNode.getChild(numChild + 1).getText()));
-              // skip the next child, since we already took care of it
-              numChild++;
-            } else if (fromNode.getChild(numChild).getType() == TOK_LIMIT) {
+                      Long.parseLong(PlanUtils.stripQuotes(fromNode.getChild(fromChildIdx + 1).getText()));
+              // Skip the next child, since we already took care of it
+              fromChildIdx++;
+            } else if (fromNode.getChild(fromChildIdx).getType() == TOK_LIMIT) {
               maxEventLimit =
-                      Integer.parseInt(PlanUtils.stripQuotes(fromNode.getChild(numChild + 1).getText()));
-              // skip the next child, since we already took care of it
-              numChild++;
+                      Integer.parseInt(PlanUtils.stripQuotes(fromNode.getChild(fromChildIdx + 1).getText()));
+              // Skip the next child, since we already took care of it
+              fromChildIdx++;
             }
             // move to the next child in FROM tree
-            numChild++;
+            fromChildIdx++;
           }
+          break;
+        }
+        default: {
+          throw new SemanticException("Unrecognized token in REPL DUMP statement.");
```

Comment:
print the token ..it will be easier to debug

Reply:
Done